agent/handlers/task_server_setup_test.go

```diff
@@ -2361,6 +2171,512 @@ func TestV4TaskMetadata(t *testing.T) {
 	})
 }
 
+func TestV2TaskMetadataWithTags(t *testing.T) {
```

Comment:
Can we do struct based testing instead for these 3 tests? I see they have similar content for each sub task, so this refactor might help reducing the duplicated code.

Reply:
We are already using struct based testing. The test is executed by `testTMDSRequest` function which is passed in a `TMDSTestCase` struct that contains the request, mock expectations, and the expected response code and body. `happyStateExpectations`, `containerInstanceTags`, `ecsInstanceTags` and other variables defined here are for reducing code duplication. 

Can you point out the duplicate code you are referring to?

Reply:
Thanks for the comment @Realmonia. We discussed offline that the three tests are already using struct based testing using `TMDSTestCase` struct. The reason we don't have a for loop is that some subtests require some setup steps which are trivial to include when using `t.Run` directly.